src/coreclr/jit/ssabuilder.cpp

```diff
@@ -743,7 +814,22 @@ void SsaBuilder::RenameDef(GenTreeOp* asgNode, BasicBlock* block)
 
     GenTreeLclVarCommon* lclNode;
     bool                 isFullDef;
-    bool                 isLocal = asgNode->DefinesLocal(m_pCompiler, &lclNode, &isFullDef);
+#if defined(TARGET_WASM)
+    bool isLocal;
+    if (block->IsLIR())
+    {
+        isLocal   = asgNode->OperIsLocalStore();
+        lclNode   = isLocal ? asgNode->AsLclVarCommon() : nullptr;
+        isFullDef = isLocal && (lclNode->OperIs(GT_STORE_LCL_VAR)
+            || (m_pCompiler->lvaTable[lclNode->GetLclNum()].lvExactSize == genTypeSize(asgNode->gtType)));
```

Comment:
I added this condition in for the size comparison as it was hitting the asserts on `GTF_VAR_DEF` below.